src/helm/benchmark/adaptation/adapters/language_modeling_adapter.py

```diff
@@ -76,7 +86,11 @@ def generate_requests(self, eval_instance: Instance) -> List[RequestState]:
 
         # Uses `max_sequence_length` instead of `max_request_length` here because `prefix_token` will be prepended
         # to the sequence later. This is the only place where `max_sequence_length` is used.
-        first_seq_len: int = min(max_sequence_length, len(tokens))
+
+        # Handle max_sequence_and_generated_tokens_length
+        first_seq_len: int = min(
+            max_sequence_length, len(tokens), max_sequence_and_generated_tokens_length - max_tokens
```

Comment:
Should this adjustment be made to `max_sequence_length` directly, rather than to `first_seq_len`?